### DISCUSSION TITLE
Report / kick / ban system

### DISCUSSION THREAD

**Discussion by samolego** [samolego/GolfIV#20] [Ideas]
This is the continuation of #3 .

What we have currently:

(These values are now configurable in config)

Suspicion value
```0 - 100 = do nothing
100 - 200 = report each X times
> 200 kick

After 10 kicks, a ban, which is not yet active.

Sus value decreases 1 per 30 seconds.

`/golf` command for inspection.

Each cheat is added to cheat list if it is not the same as previous one. Next time that player hacks their sus value is multiplied by cheat list size.

Cheat list losses 1 element (cheat) every 5 seconds.
```

**Comment by supsm** [comment]
I think sus value should decrease faster, and cheat list should decrease slower. Also as I mentioned previously, a log of some sort would be useful.

**Reply by samolego** [reply]
Log is already implemented, use `/golf player <player> seeCheatLog` command.
All of these numbers are now toggleable as well 😉 .

**Comment by spaceclouds42** [comment]
Perhaps a config option for delayed punishments. It would be similar to how Hypixel's watchdog doesn't ban you instantly, so that the cheater doesn't know what exactly got them banned. I'm not exactly sure which is the best implementation, but I thought of two ways. 

1. Every x minutes, the mod would kick everyone that had a kick value > than x minutes ago, and if they reached 10, they would be banned.
2. Maybe wait to punish until after a random delay from the time the player's kick value was incremented
